model/models/mistral3/model_text.go

```diff
@@ -16,6 +16,9 @@ type TextOptions struct {
 	hiddenSize, numHeads, numKVHeads int
 	headDim, ropeDim                 int
 	eps, ropeBase, ropeScale         float32
+	llama4Scaling                    bool
```

Comment:
you don't really need this additional flag. if `originalMaxPositionEmbeddings` (or `scalingBeta`) is zero, `getScale` should return nil. then you simply need to check if `positionScale` is nil